src/main/java/org/folio/inventory/dataimport/handlers/actions/CreateAuthorityEventHandler.java

```diff
@@ -43,8 +44,14 @@ public CreateAuthorityEventHandler(Storage storage,
     createRelationship(promise, authority, payload);
     authorityCollection.add(authority, success -> promise.complete(success.getResult()),
       failure -> {
-        LOGGER.error(String.format(FAILED_CREATING_AUTHORITY_MSG_TEMPLATE, failure.getReason(), failure.getStatusCode()));
-        promise.fail(failure.getReason());
+        //This is temporary solution (verify by error message). It will be improved via another solution by https://issues.folio.org/browse/RMB-899.
+        if (failure.getReason().contains(UNIQUE_ID_ERROR_MESSAGE)) {
```

Comment:
To be safe lets add check StringUtils.isNotBlank(failure.getReason())
Its for that cases when thrown exception has been created using constructor without message param

Reply:
@SerhiiNosko  Ok, I will add it.